sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -392,12 +463,126 @@ private[sql] class RocksDBStateStoreProvider
       case e: Throwable => throw QueryExecutionErrors.cannotLoadStore(e)
     }
   }
+
+  private object ColumnFamilyUtils {
+    private val multColFamiliesDisabledStr = "multiple column families disabled in " +
+      "RocksDBStateStoreProvider"
+
+    def getVcfIdBytes(id: Short): Array[Byte] = {
+      val encodedBytes = new Array[Byte](VIRTUAL_COL_FAMILY_PREFIX_BYTES)
+      Platform.putShort(encodedBytes, Platform.BYTE_ARRAY_OFFSET, id)
```

Comment:
`BYTE_ARRAY_OFFSET` is 0 here essentially ?

Reply:
Yes, it means 0 byte. It is the [same semantics](BYTE_ARRAY_OFFSET) as in RocksDBStateEncoder, basically it means the beginning of the encodedBytes.

Reply:
Can we mention that explicitly in the function comment ?

Reply:
We can just consider that as API spec of Platform. get/put requires starting offset and from reading byte array they require Platform.BYTE_ARRAY_OFFSET to represent the starting offset.